tach/cache/setup.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+import uuid
+from pathlib import Path
+
+
+def resolve_project_dir() -> Path:
+    current_dir = Path.cwd()
+    tach_path = current_dir / "tach.yml"
+    if not tach_path.exists():
+        raise FileNotFoundError()
+    return Path(current_dir)
```

Comment:
Should this use `find_project_config_root`? Also I think we should handle errors in the thread target (`send_log_entry` probably) since the error traces will still hit stdout otherwise.